Sources/Diagnostics/DiagnosticsTracker.swift

```diff
@@ -99,6 +99,15 @@ protocol DiagnosticsTrackerType {
                                     errorMessage: String?,
                                     errorCode: Int?,
                                     responseTime: TimeInterval)
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func trackSyncPurchasesStarted()
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func trackSyncPurchasesResult(errorMessage: String?,
+                                  errorCode: Int?,
+                                  responseTime: TimeInterval)
```

Comment:
Note that I haven't added the properties `purchases_found_count` and `unsynced_purchases_found_count`. Those were a bit trickier to track so I left them for later.